### DISCUSSION TITLE
Background Tasks

### DISCUSSION THREAD

**Discussion by matteobaldini-quix** [ionic-team/capacitor#4555] [Q&A]
I was reading the following [page](https://capacitorjs.com/docs/apis/background-task) and the first paragraph reads

> The Background Task API makes it easy to run background tasks. Currently, this plugin supports running a task when the app is backgrounded, and soon will support periodic background fetch operations.

My question is when could we expect the mentioned features? And generally is there a timeline fo future releases?

Thanks everyone

**Comment by ikeith** [comment]
I'm sorry that I don't have good news for you. [That plugin has been dropped entirely in 3.0](https://capacitorjs.com/docs/v3/updating/3-0#background-task-permissions-and-photos-plugins-removed)

> Background Task: This plugin appeared to be rarely used and didn’t quite work as most devs expected. The core team will readdress background functionality in the future. Subscribe to #3032 for updates.

As noted, what most people _really_ want is some kind of "periodic background operation", which is not what that plugin did. But that's a really hard problem to solve. Some possible approaches have been discussed but we don't have any plans to share at this time.

**Reply by matteobaldini-quix** [reply]
Damn, that is bad news. I see.

You're right that is what I wanted as well, specifically a periodic background geolocation read. Anyway, you mentioned that some possible approaches have been discussed, where could I find more info on such discussions? I really need this functionality but I've run out of ideas.

Thanks for the quick reply anyway

**Reply by pwespi** [reply]
There is a community plugin for [background-geolocation](https://github.com/capacitor-community/background-geolocation). I have never used it, but sounds like what you need.
